src/common.rs

```diff
@@ -0,0 +1,167 @@
+//! Common functionality you might need when using Specs.
+//!
+//! At the moment, this module provides two types:
+//!
+//! * `Errors`: A resource you can use to store errors that occurred outside of
+//!             the ECS but were catched inside, therefore should be handled by the user
+//!
+//! * `Merge`: A system generic over `T` which automatically merges `Ready` futures into
+//!            the component storage for `T`.
+//!
+//! To make use of these features, you need to ask for the `common` feature
+//! like this:
+//!
+//! ```toml
+//! [dependencies.specs]
+//! # version = "..."
+//! features = ["common"]
+//! ```
+
+use std::convert::AsRef;
+use std::error::Error;
+use std::fmt::{Debug, Display, Formatter, Result as FmtResult};
+use std::io::Write;
+use std::marker::PhantomData;
+
+use futures::{Async, Future};
+use {Component, Entity, Entities, FetchMut, Join, RunningTime, System, WriteStorage};
+
+/// A boxed error implementing `Debug`, `Display` and `Error`.
+pub struct BoxedErr(pub Box<Error + Send + Sync + 'static>);
```

Comment:
can we just have `type BoxedErr = ...` ? In this case simple `Box::new()` will be used instead of `BoxedErr::new()`